app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt

```diff
@@ -26,44 +46,151 @@ class SearchActivity : AppCompatActivity() {
+            updateTrackList(emptyList())
+            recyclerView.visibility = View.GONE
+            placeholderText.visibility = View.VISIBLE
+            placeholderImage.visibility = View.VISIBLE
+        }
+
+        val retrofit = Retrofit.Builder()
+            .baseUrl("https://itunes.apple.com")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        apiService = retrofit.create(ApiService::class.java)
+
+        editText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
 
             override fun afterTextChanged(s: Editable?) {
+                val searchText = s.toString()
+                clearButton.visibility = if (searchText.isNotEmpty()) View.VISIBLE else View.GONE
+
+                if (searchText.isNotEmpty()) {
+                    searchTracks(searchText)
+                } else {
+                    updateTrackList(emptyList())
+                    placeholderText.visibility = View.VISIBLE
+                    placeholderImage.visibility = View.VISIBLE
+                    recyclerView.visibility = View.GONE
+                    if (!isRefreshing) {
+                        refreshButton.visibility = View.GONE
+                    }
+                }
             }
         })
 
-        clearButton.setOnClickListener {
-            editText.text.clear()
-            hideKeyboard()
+        refreshButton.setOnClickListener {
+            val searchText = editText.text.toString()
+            if (searchText.isNotEmpty()) {
+                searchTracks(searchText)
+            }
         }
 
-        savedText = savedInstanceState?.getString("savedText")
-        savedText?.let { editText.setText(it) }
+        if (savedInstanceState != null) {
+            savedText = savedInstanceState.getString("savedText", "")
+            editText.setText(savedText)
+            if (savedText.isNotEmpty()) {
+                searchTracks(savedText)
+            }
+        }
     }
 
-
     override fun onSaveInstanceState(outState: Bundle) {
         super.onSaveInstanceState(outState)
         outState.putString("savedText", editText.text.toString())
     }
+
     private fun hideKeyboard() {
         val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
         imm.hideSoftInputFromWindow(editText.windowToken, 0)
     }
-    override fun onResume() {
-        super.onResume()
-        trackAdapter.run { notifyDataSetChanged() }
+
+    private fun searchTracks(keyword: String) {
+        sendRequest(keyword)
+    }
+
+    private fun sendRequest(keyword: String) {
+        apiService.search(keyword)?.enqueue(object: Callback<Tracks?> {
+            override fun onResponse(call: Call<Tracks?>, response: Response<Tracks?>) {
+                if (response.code() == 200) {
```

Comment:
```suggestion
                if (response.isSuccessfull) {
```
Чтобы не хардкодить коды ответов http, можно воспользоваться свойством `isSuccessfull`